### ISSUE TITLE
FLS to cloudcompare

### ISSUE THREAD

**Issue by GManduwi** [cloudcompare/trunk#197]
I am trying to bring in FLS files into CC and am having problems viewing the correct colours in CC. If I export the same FLS file to a .pts file, then bring it into CC, the points have the correct colour values (RGB). How do I manage this from import of a .fls file?

**Comment by cloudcompare** [comment]
I believe the FARO filter only imports the 'intensity' value (i.e. a scalar field) and not the RGB colors.

I don't even think I have FLS files with RGB colors to test this ;).

**Comment by GManduwi** [comment]
So if I needed to decimate large pts files in CC I would have to first export them to .pts in Faro Scene, then bring them into CC and subsample them  so I can keep the RGB values?

**Comment by cloudcompare** [comment]
Right now yes. Or you could use the E57 format which should be more efficient.

And if you can send me fls samples with RGB colors, I'll gladly (try to) update the FARO filter.

**Comment by GManduwi** [comment]
Ok. I have quite a few of them , but all around 200MB, limit on here is 10MB.

**Comment by cloudcompare** [comment]
Yes that's generally the issue ;)

Do you have a dropbox or Google drive account ? You can send me emails at 'cloudcompare [at] danielgm.net'

**Comment by kbazley** [comment]
I have Dropbox, so I can send you an .fls file.

**Comment by cloudcompare** [comment]
Yes, any FLS file with RGB data would be great!

**Comment by davidcanada** [comment]
I am getting a load error when opening a .fls file.  The Faro i/o plugin was installed by checking the Faro option in the CloudCompare installer.  Is qFaro_i/o supposed to be listed in the plugin list ? (not seeing it). This is the message: 
![cc faro scan opening error](https://cloud.githubusercontent.com/assets/7146244/7169403/551aa3d2-e382-11e4-93fd-07da1a40eb02.JPG)

**Comment by cloudcompare** [comment]
Nope, in your case it seems that the plugin and the FARO LS runtime were successfully loaded (see the 2nd message in the Console).

I also have a file on my side that has a similar problem. It's a fls file that was copied from a project folder without the remaining files. It appears that the Faro LS runtime looks for the workspace/project file first before actually loading the scan files...

**Comment by MtnGraphix** [comment]
Has anyone found an answer to this?   ... trying to avoid Scene as much as possible.. Love CloudCompare but I would like to bring in the fls files with color...anyone?

**Comment by MtnGraphix** [comment]
wow! .. thank you for such a quick reply! .. I appreciate CloudCompare more and more everyday! ... When I scan using my x330 the result is a .fls folder containing an .fls file... Is there a preferred method to bring this fls into CC... I've tried dragging it in as well as opening from the File menu... I am asked if I want the scan imported with color.. I always say Yes, however it always comes in as black and white ( I do know that I did scan it with collect color option on) .. thank you for your help!

**Comment by cloudcompare** [comment]
Oh then it's potentially another issue. FARO files can store the color not as RGB info but as separate pictures (as taken by the embedded camera). In this case the FARO LS SDK can't output RGB color per point... And FARO didn't answer me about how other (pro) software were able to extract the color nevertheless...

**Comment by MtnGraphix** [comment]
Thank you very much for trying to help!

**Comment by Terra3D** [comment]
Does anyone have news about getting the rgb values on importing the Faro fls to cc? We use a workaround via Autodesk recap at the moment. So somewhere the color information is possibly stored in the fls structure. We tried with other software as well, but they all probably rely on the Faro SDK with the same greyscale results.

**Comment by dgirardeau** [comment]
I never got any detail / answer about this from FARO. It will be hard for me to do it without clear instructions.

**Comment by whatnick** [comment]
I would like some work on this as well. Just received some sample Faro files which work with their chromimum based (WebGL?) viewer. Will explore the JS codepaths there.

**Comment by dgirardeau** [comment]
See http://www.cloudcompare.org/forum/viewtopic.php?f=9&t=1991&p=11717#p11717
